pinot-segment-local/src/main/java/org/apache/pinot/segment/local/segment/creator/impl/SegmentIndexCreationDriverImpl.java

```diff
@@ -157,17 +159,20 @@ public RecordReader getRecordReader() {
   public void init(SegmentGeneratorConfig config, RecordReader recordReader)
       throws Exception {
     SegmentCreationDataSource dataSource = new RecordReaderSegmentCreationDataSource(recordReader);
-    init(config, dataSource, new TransformPipeline(config.getTableConfig(), config.getSchema()));
+    init(config, dataSource, RecordEnricherPipeline.fromTableConfig(config.getTableConfig()),
+        new TransformPipeline(config.getTableConfig(), config.getSchema()));
```

Comment:
Does the `RecordEnricherPipeline` evaluate before the `TransformPipeline` or after? If it's after then we should have it occur after in the ctor parameter list, so that the APIs help re-enforce the logical behavior.

Reply:
Record enricher would evalute before the transform pipeline